retrievalmarket/impl/provider_test.go

```diff
@@ -637,17 +645,49 @@ func TestDynamicPricing(t *testing.T) {
 	}
 }
 
+func makeIdentityCidWith(cids []cid.Cid, padding ...[]byte) (cid.Cid, error) {
+	node, err := qp.BuildList(basicnode.Prototype.List, int64(len(cids)+len(padding)), func(la datamodel.ListAssembler) {
+		for _, cid := range cids {
+			qp.ListEntry(la, qp.Link(cidlink.Link{Cid: cid}))
+		}
+		for _, pad := range padding {
+			qp.ListEntry(la, qp.Bytes(pad))
+		}
+	})
+	if err != nil {
+		return cid.Undef, err
+	}
+	encoded, err := ipld.Encode(node, dagcbor.Encode)
+	if err != nil {
+		return cid.Undef, err
+	}
+	lp := cidlink.LinkPrototype{Prefix: cid.Prefix{Version: 1, Codec: 0x71, MhType: 0x0}}
```

Comment:
I think we're all good for UnixFS, but we had better add a test case since it's the default one we have an issue with.

Reply:
added some dag-pb variants to this too